### PR TITLE
⚠️ Remove migration to new VPN filters

### PR DESCRIPTION
**Pull request by axel7born**
**What this PR does / why we need it**:
Remove the code, that migrates shoots from the old shared VPN filter to the new shoot specific filter.
Once the migration is done for all shoots, the migrations is not needed any longer.
For new shoots, the migration code leads to the creation of the shared filter, which will be removed after the first reconcile. This is not a problem, as long as the acl-extension is not enabled for a bigger number of shoots at once. In this case , the shared filter could become too large.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

ref https://github.com/stackitcloud/gardener-extension-acl/pull/43
Also see the corresponding ADR: https://github.com/stackitcloud/gardener-extension-acl/blob/e69b82c1a4a9a98e95238d6600a0642f55a4356f/docs/adr/06_seperate_vpn_filter.md

**Release Note**:

> ⚠️ Before rolling out this version, ensure that you have rolled out version [v1.3.0](https://github.com/stackitcloud/gardener-extension-acl/releases/tag/v1.3.0) of the ACL extension. Ensure that all shoots using the ACL extension have been reconciled, so that all legacy `EnvoyFilter` objects named `acl-vpn` have been removed from all seed clusters, and the shoot-specific `acl-vpn-shoot-*` `EnvoyFilter` objects have been created to replace them.

### CODE REVIEW

**Review by timebertt — COMMENTED** [review]
Thanks @axel7born, appreciated!
I was wondering whether we should do a breaking change and directly remove the migration code instead of putting it behind a toggle.
We want to remove the migration code later on anyhow and operators will need to ensure that all shoots have been migrated.
Would it be reasonable to do it with this PR, i.e., the next release?

**Comment by axel7born** [comment]
@timebertt Could you please take a look again?

**Comment by robinschneider** [comment]
> @timebertt Could you please take a look again?

He will get back to you in about two weeks, sorry for the delay 😕

**Comment by timebertt** [comment]
Back from vacation, sorry for the delay

File: pkg/controller/actuator.go [hunk 1]
```diff
@@ -303,58 +291,6 @@ func (a *actuator) Migrate(ctx context.Context, log logr.Logger, ex *extensionsv
...
-	if len(aclMappings) == 0 {
-		// no shoot in this namespace with the ACL extension, so we can delete the config
-		err = a.client.Delete(ctx, envoyFilter)
-		return client.IgnoreNotFound(err)
-	}
-	if istioLabels == nil {
-		istioLabels = istioLabelsFromExt
-	}
-
-	vpnEnvoyFilterSpec, err := envoyfilters.BuildLegacyVPNEnvoyFilterSpecForHelmChart(
```
[the remark this answers is not in the document]
Comment (axel7born) [reply pkg/controller/actuator.go:337]: Removed.

**Comment by axel7born** [comment]
Thanks for the review! The release note is perfect. :+1:

File: pkg/envoyfilters/envoyfilters.go [hunk 2]
```diff
@@ -94,32 +94,6 @@ func BuildVPNEnvoyFilterSpecForHelmChart(
...
-// shoot - this doesn't work because all the VPN traffic flows through the same
-// filter.
-//
-// We use the technical ID of the shoot for the VPN rule, which is de facto the
-// same as the seed namespace of the shoot. (Gardener uses the seedNamespace
-// value in the botanist vpnshoot task.)
-func BuildLegacyVPNEnvoyFilterSpecForHelmChart(
-	mappings []ACLMapping, alwaysAllowedCIDRs []string, istioLabels map[string]string,
-) (map[string]interface{}, error) {
-	vpnConfigPatch, err := CreateLegacyVPNConfigPatchFromRule(mappings, alwaysAllowedCIDRs)
```
Comment (timebertt) [inline pkg/envoyfilters/envoyfilters.go:108]: Now, `CreateLegacyVPNConfigPatchFromRule` is unused 😄

**Review by timebertt — APPROVED** [review]
I pushed one more commit to drop more unused code.
Thanks again for the cleanup!

**Pull request merged**